js/iframe-coordinator/src/HostRouter.ts

```diff
@@ -1,97 +1,73 @@
-import ClientFrame from './elements/x-ifc-frame';
-import {
-  ClientToHost,
-  validate as validateIncoming
-} from './messages/ClientToHost';
-import {
-  HostToClient,
-  validate as validateOutgoing
-} from './messages/HostToClient';
-import { Publication } from './messages/Publication';
-
 interface ClientRegistration {
   url: string;
   assignedRoute: string;
```

Comment:
I have wondered why we need both the assignedRoute and the key in the RoutingMap.  From what I recall, people would changeRoute using the assignedRoute, so the key was useless.

Could we either changeRoute based of the key and then reuse url for both a full url and Route.  Or just use a straight array for the ClientRegistrations?

Reply:
Giving each client a unique Id will help us when we integrate things like logging, so there's a logical namespace to log under for each client. It may even be a good idea to tag all messages to the host in general with the id of the client they originate from, so the host can react differently based on the client, if they need to.

Reply:
Makes sense.  As long as there was a purpose behind it. I like the idea of putting in on the events as well.   Probably worth doing once we re-visit the subscription/eventing APIs.

Reply:
One thing I was thinking about while I was adding documentation.  I would like to consider changing changeRoute to use the id instead of the route.  Or have an override which takes that.

At least in PureConnect we will only use urls.  So it would be great if we could just provide the id and the url.  And besides, using an id to change the route seems much more inline with what you had in mind for ids.